ui/panels-plugin/src/plugins/time-series-chart/VisualOptionsEditor.tsx

```diff
@@ -35,24 +40,38 @@ export function VisualOptionsEditor({ value, onChange }: VisualOptionsEditorProp
   const handleLineWidthChange = (_: Event, sliderValue: number | number[]) => {
     const newValue = Array.isArray(sliderValue) ? sliderValue[0] : sliderValue;
     const symbolSize = newValue !== undefined ? newValue + POINT_SIZE_OFFSET : DEFAULT_POINT_RADIUS;
-    onChange({
-      ...value,
-      lineWidth: newValue,
-      pointRadius: symbolSize,
-    });
+    onChange(
+      produce(value, (draft) => {
+        draft.lineWidth = newValue;
+        draft.pointRadius = symbolSize;
+      })
+    );
   };
 
   const handleAreaOpacityChange = (_: Event, sliderValue: number | number[]) => {
     const newValue = Array.isArray(sliderValue) ? sliderValue[0] : sliderValue;
-    onChange({
-      ...value,
-      areaOpacity: newValue,
-    });
+    onChange(
+      produce(value, (draft) => {
+        draft.areaOpacity = newValue;
+      })
+    );
   };
 
   const currentStack: StackOptions = value.stack ?? 'none';
   const stackConfig = STACK_CONFIG[currentStack];
 
+  const handleSingleSeriesColorChange = (color: string) => {
+    onChange(
+      produce(value, (draft) => {
+        draft.palette = {
+          ...draft.palette,
+          mode: 'auto',
```

Comment:
TODO: Need to remove ‘auto’ here and always fallback to whatever is set in palette.mode 

Planning to fix this and I’ll add a test case for ‘categorical’ as fallback (when singleSeriesColor is set).
Also would love to add visual regression / an e2e for this but not sure I’ll have enough time as part of this PR